### PR TITLE
Pde remove shapefunctions

### PR DESCRIPTION
**Pull request by pmueller2**
This basically continues work related to #210. Shape functions are moved from `ElementShapeFunctions` to the interpolation classes as static functions and renamed. It is done in the same way as for the linear triangle already in branch 23b8c23a also referenced in the issue.

### CODE REVIEW

File: nuto/mechanics/interpolation/InterpolationTriangleLinear.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+    derivativeShapeFunctions(1, 0) = 1.0;
+    derivativeShapeFunctions(1, 1) = 0.0;
+
+    derivativeShapeFunctions(2, 0) = 0.0;
+    derivativeShapeFunctions(2, 1) = 1.0;
+    return derivativeShapeFunctions;
+}
+
+//////////////////////////////////////////////////////////////////////////////////////////////////////////////
```
Comment (Psirus) [inline nuto/mechanics/interpolation/InterpolationTriangleLinear.cpp]: Copy&paste remainder.

File: nuto/mechanics/interpolation/TypeDefs.h [hunk 2]
```diff
@@ -7,19 +7,18 @@ namespace NuTo
 
 typedef Eigen::VectorXd NodeValues;
 // typedef Eigen::VectorXd NaturalCoords;
-typedef Eigen::VectorXd ShapeFunctions;
 typedef Eigen::MatrixXd DerivativeShapeFunctionsNatural;
 typedef Eigen::MatrixXd DerivativeShapeFunctionsGlobal;
 typedef Eigen::MatrixXd BMatrixStrain;
 typedef Eigen::MatrixXd BMatrixGradient;
 typedef Eigen::MatrixXd NMatrix;
```
Comment (Psirus) [inline nuto/mechanics/interpolation/TypeDefs.h]: Can we not get rid of all of these, except for `NaturalCoords` and `GlobalCoords`?

**Review by Psirus — APPROVED** [review]
Nice, thanks for this labour-intensive stuff.

Also, I was wondering if it would make sense to make the `Shape` of the interpolation type public, so that you could get the actual type in case you wanted it, instead of `Shape&`?

File: nuto/mechanics/interpolation/TypeDefs.h (hunk 2 above)
Comment (pmueller2) [reply nuto/mechanics/interpolation/TypeDefs.h]: I will remove them.

File: nuto/mechanics/cell/Jacobian.h [hunk 3]
```diff
@@ -54,43 +53,43 @@ class Jacobian
...
+                Eigen::Matrix<double, 3, 2> jacobian = CalculateFixedSize<3, 2>(nodeValues, derivativeShapeFunctions);
+                Eigen::Matrix<double, 3, 3> extendedJacobian;
                 mNormal = (jacobian.col(0)).cross(jacobian.col(1)).normalized();
                 extendedJacobian.col(0) = jacobian.col(0);
                 extendedJacobian.col(1) = jacobian.col(1);
                 extendedJacobian.col(2) = mNormal;
                 mJacobian = jacobian;
-                mInvJacobian = extendedJacobian.inverse().block<2,3>(0,0);;
+                mInvJacobian = extendedJacobian.inverse().block<2, 3>(0, 0);
+                ;
```
Comment (Psirus) [inline nuto/mechanics/cell/Jacobian.h:76]: Superfluous semicolon.

File: nuto/mechanics/cell/Jacobian.h [hunk 4]
```diff
@@ -54,43 +53,43 @@ class Jacobian
...
             default:
             {
                 throw;
             }
             }
-        } else
+        }
+        else
         {
             throw;
```
Comment (Psirus) [inline nuto/mechanics/cell/Jacobian.h:88]: What is going on here? Why don't these give at least some indication what's wrong? I know this was not part of the PR, I just noticed this now.

Comment (pmueller2) [reply nuto/mechanics/cell/Jacobian.h:88]: Will be addressed in another PR.
